staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -83,3 +93,54 @@ func (record *attributesRecord) GetOldObject() runtime.Object {
 func (record *attributesRecord) GetUserInfo() user.Info {
 	return record.userInfo
 }
+
+func (record *attributesRecord) GetAnnotations() map[string]string {
+	record.annotationLock.RLock()
+	defer record.annotationLock.RUnlock()
+
+	if record.annotations == nil {
+		return nil
+	}
+	cp := make(map[string]string)
+	for key, value := range record.annotations {
+		cp[key] = value
+	}
+	return cp
+}
+
+func (record *attributesRecord) SetAnnotations(annotations map[string]string) {
+	record.annotationLock.Lock()
+	defer record.annotationLock.Unlock()
+
+	for key, value := range annotations {
+		record.setAnnotation(key, value, false)
+	}
+}
+
+func (record *attributesRecord) SetAnnotation(key, value string) {
+	record.setAnnotation(key, value, true)
+}
+
+func (record *attributesRecord) setAnnotation(key, value string, lock bool) {
+	parts := strings.Split(key, "/")
+	if len(parts) != 2 {
+		glog.Warningf("Annotation key has invalid format, the right format is a DNS subdomain prefix and '/' and key name. (e.g. 'podsecuritypolicy.admission.k8s.io/admit-policy')")
```

Comment:
why is this not an error we pass down to the caller?

Reply:
Done.